daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/DelimiterParsers.scala

```diff
@@ -73,6 +76,35 @@ class DelimiterTextParser(
     foundLocalDFAIndex >= 0
   }
 
+  private def localDelimiters(state: PState): Seq[DFADelimiter] = {
+    var foundLocalDFAIndex = -1
+    val localIndexStart = state.mpstate.delimitersLocalIndexStack.top
+    val inScopeDelimiters = state.mpstate.delimiters
+    val res = inScopeDelimiters.slice(localIndexStart, inScopeDelimiters.length)
+    res
+  }
+
+  private def didNotFindExpectedDelimiter(foundDelimiter: ParseResult, start: PState): Unit = {
+    val delimString = foundDelimiter.matchedDelimiterValue.get
+    val localDelims = localDelimiters(start)
+    val foundDFA = foundDelimiter.matchedDFAs(0)
+    PE(
+      start,
+      s"""Found enclosing delimiter: %s during scan for local delimiter(s): %s.
+         | The expected delimiter(s) were: %s.
+         | The enclosing delimiter was from %s.
+         |""".stripMargin,
+      foundDFA.strForDiagnostic,
+      localDelims.map { d => d.strForDiagnostic }.mkString(", "),
+      localDelims
+        .map { d =>
+          s"  ${d.delimType.toString} ${d.strForDiagnostic} from location ${d.location}."
+        }
+        .mkString("\n", "\n", ""),
+      foundDFA.location,
```

Comment:
Thoughts on including `locationDescription` either instead of or in addition to `.location`? `.location` just outputs the element name which can sometimes make it hard to figure out where in the schema it actually is. `locationDescription` includes file/line/col info so makes it real easy.

Reply:
Ah. I was wondering why I wasn't getting file/line num info. I though it might just be due to TDML tests with embedded schema.

Will fix.